tests/ops/op_math/test_controlled_decompositions.py

```diff
@@ -241,6 +241,47 @@ def test_trivial_ops_in_decomposition(self):
         assert len(decomp) == 5
         assert decomp == expected
 
+    @pytest.mark.parametrize(
+        "composite_op, want_decomp",
+        [
+            (
+                qml.ops.Prod(qml.PauliX(0), qml.PauliX(0)),  # type: ignore
+                [
+                    qml.CNOT(wires=[1, 0]),
+                    qml.CNOT(wires=[1, 0]),
+                ],
+            ),
+            (
+                qml.ops.SProd(1.23, qml.PauliX(0)),  # type: ignore
+                [
+                    qml.RZ(np.pi / 2, wires=0),
+                    qml.RY(np.pi / 2, wires=0),
+                    qml.CNOT(wires=[1, 0]),
+                    qml.RY(-np.pi / 2, wires=0),
+                    qml.RZ(-2 * np.pi, wires=0),
+                    qml.CNOT(wires=[1, 0]),
+                    qml.RZ(3 * np.pi / 2, wires=0),
+                ],
+            ),
+            (
+                qml.ops.Sum(qml.PauliX(0), qml.PauliX(0)),  # type: ignore
```

Comment:
```
                qml.s_prod(1/np.sqrt(2), qml.X(0)) + qml.s_prod(1/np.sqrt(2), qml.Y(0)),
```
Similar to the comment above.  It might be worth running the tests on unitary targets, and this would be a sum object that would still be unitary.

Reply:
Agreed!